src/middlewared/middlewared/plugins/afp.py

```diff
@@ -39,3 +43,196 @@ class Config:
             await self._update_service(old, new)
 
         return new
+
+
+class SharingAFPService(CRUDService):
+    class Config:
+        namespace = 'sharing.afp'
+        datastore = 'sharing.afp_share'
+        datastore_prefix = 'afp_'
+        datastore_extend = 'sharing.afp.extend'
+
+    @accepts(Dict(
+        'sharingafp_create',
+        Str('path'),
+        Bool('home'),
+        Str('name'),
+        Str('comment'),
+        List('allow'),
+        List('deny'),
+        List('ro'),
+        List('rw'),
+        Bool('timemachine'),
+        Int('timemachine_quota'),
+        Bool('nodev'),
+        Bool('nostat'),
+        Bool('upriv'),
+        Int('fperm', validators=[Range(min=0, max=0o777)], default=0o644),
```

Comment:
In FTP plugin we have masks passed as strings: `Str('filemask', validators=[Match(r"^[0-7]{3}$")])`. Here they are integers. This is API inconsistency.

I have no idea what is better, but we should choose one way. I myself don't like integers because we have JSON API and JSON integers can only have decimal representation which can cause confusion if user would interact with API directly or even with `midclt`.

Anyway, let's have an `UnixMask` and `UnixPermission()` schema because it's widely used in FreeNAS.

Reply:
Sorry, I've just looked through @william-gr 's previous review and he'd asked for opposite thing :)

So this is more a subject to discuss than changes request.

Reply:
I do like the idea of `UnixMask` and `UnixPermission`.

I myself prefer integers because thats why they are at essence.

How about we accept both as input using these new schemas but always output integer?

Reply:
@william-gr then `midclt call afp.config` will output e.g. `"fperm": 420` for 0644, is this acceptable?

Reply:
@themylogin I personally dont have a problem with that, but I see what you're saying. Sounds like simply using string would avoid confusion : ).

@skarekrow will hate me : D

Reply:
@william-gr another (probably not valid) reason for using strings is that new UI permissions component is already using strings https://github.com/freenas/webui/blob/c888489c452f40e3e482de49a667d6558e2b7ec1/src/app/pages/common/entity/entity-form/components/form-permissions/form-permissions.component.ts

Reply:
Fair enough, strings everywhere! Sorry brandon :D